File: MenuOperaciones/app/src/main/java/com/example/menuoperaciones/MainActivity.java
package com.example.menuoperaciones;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {
    Button btnSuma;
    Button btnDivision;
    Button btnRCuadrada;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //enlaces
        btnSuma = (Button) findViewById(R.id.btnSuma);
        btnDivision = (Button) findViewById(R.id.btnDivision);
        btnRCuadrada = (Button) findViewById(R.id.btnRCuadrada);

        //oyentes
        btnSuma.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                abrirSuma();
            }
        });

        btnDivision.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) { abrirDivision(); }
        });

        btnRCuadrada.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) { abrirRCuadrada(); }
        });
    }

    private void abrirSuma(){
        Intent intent1 = new  Intent(getApplicationContext(), SumaActivity.class);
        startActivity(intent1);
    }

    private void abrirDivision(){
        Intent intent2 = new Intent(getApplicationContext(), DivisionActivity.class);
        startActivity(intent2);
    }

    private void abrirRCuadrada(){
        Intent intent3 = new Intent(getApplicationContext(), RCuadradaActivity.class);
        startActivity(intent3);
    }
}

File: Chatbot/chatbotApp/app/src/main/java/com/example/chatbotapp/MainActivity.kt
package com.example.chatbotapp

import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


class MainActivity : AppCompatActivity() {
    private val  adapterChatBot = AdapterChatBot()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val retrofit=Retrofit.Builder()
            .baseUrl("http:192.168.8.101:5000/")
            .addConverterFactory(GsonConverterFactory.create())
            .build()


        val apiService = retrofit.create(APIService::class.java)



        rvChatList.layoutManager = LinearLayoutManager(this)
        rvChatList.adapter = adapterChatBot

        btnSend.setOnClickListener{
            if(etChat.text.isNullOrEmpty()){
                Toast.makeText(this@MainActivity, "Por favor escriba algo", Toast.LENGTH_LONG).show()
                return@setOnClickListener
            }
            adapterChatBot.addChatToList(ChatModel(etChat.text.toString()))
            apiService.chatWithTheBit(etChat.text.toString()).enqueue(callBack)
                    etChat.text.clear()

        }

    }
       private val callBack = object : Callback<ChatResponse>{
           override fun onResponse(call: Call<ChatResponse>, response: Response<ChatResponse>) {
               if (response.isSuccessful &&  response.body()!=null) {

                   adapterChatBot.addChatToList(ChatModel(response.body()!!.chatBotReply, true))

               }else{
                   Toast.makeText(this@MainActivity, "hubo algun error", Toast.LENGTH_LONG).show()
               }
           }

           override fun onFailure(call: Call<ChatResponse>, t: Throwable) {
               Toast.makeText(this@MainActivity, "hubo algun error", Toast.LENGTH_LONG).show()

           }

       }
}